onnxruntime/test/providers/cpu/nn/group_norm_op_test.cc

```diff
@@ -0,0 +1,150 @@
+    test.AddOutput<TypeParam>("output", dims, GetTypedArray<TypeParam>({-1.4638f, -0.8783f, -0.2928f, 0.2928f, 0.8783f, 1.4638f}));
+
+    std::vector<std::unique_ptr<IExecutionProvider>> execution_providers;
+    execution_providers.push_back(DefaultCoreMLExecutionProvider(true));
+    // coreml EP requires weight and bias to be initializers
+    test.Run(OpTester::ExpectResult::kExpectSuccess, "", {}, nullptr, &execution_providers);
+  };
+
+  // gamma as initialized will fail for CPUEP
+  run_test(true);
+}
+
+TYPED_TEST(GroupNormalizationOpTest, groupsizen) {
+  OpTester test("GroupNormalization", 18);
+  test.AddAttribute("epsilon", 0.3F);
+  test.AddAttribute("num_groups", int64_t(2));
+
+  vector<float> input = {-1.1258f, -1.1524f, -0.2506f, -0.4339f,
+                         0.8487f, 0.6920f, -0.3160f, -2.1152f,
+                         0.3223f, -1.2633f, 0.3500f, 0.3081f,
+                         0.1198f, 1.2377f, 1.1168f, -0.2473f,
+                         -1.3527f, -1.6959f, 0.5667f, 0.7935f,
+                         0.5988f, -1.5551f, -0.3414f, 1.8530f,
+
+                         0.7502f, -0.5855f, -0.1734f, 0.1835f,
+                         1.3894f, 1.5863f, 0.9463f, -0.8437f,
+                         -0.6136f, 0.0316f, -0.4927f, 0.2484f,
+                         0.4397f, 0.1124f, 0.6408f, 0.4412f,
+                         -0.1023f, 0.7924f, -0.2897f, 0.0525f,
+                         0.5229f, 2.3022f, -1.4689f, -1.5867f};
+  vector<int64_t> input_dims = {2, 6, 4};
+  test.AddInput<TypeParam>("X", input_dims, GetTypedArray<TypeParam>(input));
+
+  vector<float> scale = {1.F, 1.F, 1.F, 1.F, 1.F, 1.F};
+  vector<int64_t> scale_dims = {6};
+  test.AddInput<TypeParam>("scale", scale_dims, GetTypedArray<TypeParam>(scale), true);
+
+  vector<float> B = {.0F, .0F, .0F, .0F, .0F, .0F};
+  vector<int64_t> B_dims = {6};
+  test.AddInput<TypeParam>("bias", B_dims, GetTypedArray<TypeParam>(B), true);
+
+  vector<float> expected_output = {
+      -0.7590f, -0.7848f, 0.0914f, -0.0867f,
+      1.1595f, 1.0073f, 0.0278f, -1.7203f,
+      0.6480f, -0.8926f, 0.6749f, 0.6343f,
+      0.0232f, 0.9274f, 0.8296f, -0.2738f,
+      -1.1679f, -1.4456f, 0.3846f, 0.5681f,
+      0.4107f, -1.3317f, -0.3499f, 1.4252f,
+
+      0.5772f, -0.8298f, -0.3957f, -0.0198f,
+      1.2505f, 1.4580f, 0.7838f, -1.1017f,
+      -0.8594f, -0.1798f, -0.7320f, 0.0486f,
+      0.2541f, -0.0377f, 0.4334f, 0.2554f,
+      -0.2291f, 0.5686f, -0.3962f, -0.0911f,
+      0.3282f, 1.9145f, -1.4475f, -1.5525f};
+  test.AddOutput<TypeParam>("Y", input_dims, GetTypedArray<TypeParam>(expected_output));
+
+  std::vector<std::unique_ptr<IExecutionProvider>> execution_providers;
+  execution_providers.push_back(DefaultCoreMLExecutionProvider(true));
+  // coreml EP requires weight and bias to be initializers
+  if constexpr (std::is_same<TypeParam, float>::value) {
+    test.SetOutputTolerance(1e-4f);
+  } else {
+    test.SetOutputTolerance(0.005f);
```

Comment:
nit: Can we do this by checking for MLFloat16 as we only want to lower the tolerance in that case don't we?

Reply:
I used random input values instead of values like (1,2,4,5,6,7), thus the error of float dtype is large than the default.